include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -0,0 +1,147 @@
+                        BlockSharedMemStMemberImpl(BlockSharedMemStMemberImpl const &) = delete;
+                        //-----------------------------------------------------------------------------
+                        BlockSharedMemStMemberImpl(BlockSharedMemStMemberImpl &&) = delete;
+                        //-----------------------------------------------------------------------------
+                        auto operator=(BlockSharedMemStMemberImpl const &) -> BlockSharedMemStMemberImpl & = delete;
+                        //-----------------------------------------------------------------------------
+                        auto operator=(BlockSharedMemStMemberImpl &&) -> BlockSharedMemStMemberImpl & = delete;
+                        //-----------------------------------------------------------------------------
+                        /*virtual*/ ~BlockSharedMemStMemberImpl() = default;
+
+                        template <typename T>
+                        void alloc() const
+                        {
+                            uint8_t* buf = &m_mem[m_allocdBytes];
+                            new (buf) T();
+                            m_allocdBytes += alignPitch<T>();
+#if (defined ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST) && (! defined NDEBUG)
+                            ALPAKA_ASSERT(m_allocdBytes < m_capacity);
+#endif
+                        }
+
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wcast-align" // "cast from 'unsigned char*' to 'unsigned int*' increases required alignment of target type"
+#endif
+                        template <typename T>
+                        T& getLatestVar() const
+                        {
+                           return *reinterpret_cast<T*>(&m_mem[m_allocdBytes-alignPitch<T>()]);
+                        }
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic pop
+#endif
+
+                        void free() const
+                        {
+                            m_allocdBytes = 0u;
+                        }
+
+                    private:
+                        mutable unsigned int m_allocdBytes = 0;
+                        mutable uint8_t* m_mem;
+#ifndef NDEBUG
+                        const unsigned int m_capacity;
+#endif
+
+                        template<typename T>
+                        static constexpr size_t alignPitch()
+                        {
+                            return (sizeof(T)/TDataAlignBytes + (sizeof(T)%TDataAlignBytes>0))*TDataAlignBytes;
+                        }
+                    };
+                }
+                //#############################################################################
+                //! Static block shared memory provider using a pointer to
+                //! externally allocated fixed-size memory, likely provided by
+                //! BlockSharedMemDynMember.
+                template<unsigned int TDataAlignBytes = core::vectorization::defaultAlignment*8>
```

Comment:
Shouldn't it be  `core::vectorization::defaultAlignment/8`?

Reply:
Ups, yes.